webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/MeshFileService.scala

```diff
@@ -368,24 +377,46 @@ class MeshFileService @Inject()(config: DataStoreConfig)(implicit ec: ExecutionC
   def readMeshChunkV3(organizationName: String,
                       dataSetName: String,
                       dataLayerName: String,
-                      meshChunkDataRequest: MeshChunkDataRequestV3,
-  ): Fox[(Array[Byte], String)] = {
-    val meshFilePath = dataBaseDir
-      .resolve(organizationName)
-      .resolve(dataSetName)
-      .resolve(dataLayerName)
-      .resolve(meshesDir)
-      .resolve(s"${meshChunkDataRequest.meshFile}.$meshFileExtension")
-
-    executeWithCachedHdf5(meshFilePath, meshFileCache) { cachedMeshFile =>
-      val meshFormat = cachedMeshFile.reader.string().getAttr("/", "mesh_format")
-      val data =
-        cachedMeshFile.reader
-          .uint8()
-          .readArrayBlockWithOffset("neuroglancer", meshChunkDataRequest.byteSize, meshChunkDataRequest.byteOffset)
-      (data, meshFormat)
-    } ?~> "mesh.file.readData.failed"
-  }
+                      meshChunkDataRequests: MeshChunkDataRequestV3List,
+  ): Fox[(Array[Byte], String)] =
+    for {
+      data: List[(Array[Byte], String, Int)] <- Fox.combined(
+        meshChunkDataRequests.requests.zipWithIndex
+          .sortBy(requestAndIndex => requestAndIndex._1.byteOffset)
+          .map(requestAndIndex => {
+            val meshChunkDataRequest = requestAndIndex._1
+            val meshFilePath = dataBaseDir
+              .resolve(organizationName)
+              .resolve(dataSetName)
+              .resolve(dataLayerName)
+              .resolve(meshesDir)
+              .resolve(s"${meshChunkDataRequests.meshFile}.$meshFileExtension")
+
+            executeWithCachedHdf5(meshFilePath, meshFileCache) { cachedMeshFile =>
+              val meshFormat = cachedMeshFile.reader.string().getAttr("/", "mesh_format")
+              val data =
+                cachedMeshFile.reader
+                  .uint8()
+                  .readArrayBlockWithOffset("neuroglancer",
+                                            meshChunkDataRequest.byteSize,
+                                            meshChunkDataRequest.byteOffset)
+              (data, meshFormat, requestAndIndex._2)
+            } ?~> "mesh.file.readData.failed"
+          })
+          .toList)
+      dataSorted = data.sortBy(d => d._3)
+      _ <- Fox.bool2Fox(data.map(d => d._2).toSet.size == 1) // Ensure same encoding for all responses
+      encoding = data.map(d => d._2).head
+      jumpTable = dataSorted
+        .map(d => d._1.length)
+        // The jump table contains n + 1 elements (that way, for each requested
+        // chunk i , jumpTable[i] and jumpTable[i + 1] define the byte range).
+        // The jumptable contains the offsets in bytes (encoded as Longs).
+        .scanLeft((8 * (data.length + 1)).toLong)((a, b) => a + b)
+        .flatMap(l => longToBytes(l))
```

Comment:
@frcroth I adapted the jump table so that its entries are in byte and not in elements (in other words, I added the `8 * ...` part in line 415). Probably the "magic" 8 can be expressed by retrieving the byte count of a `Long`? Could you tidy this up?
Also, I have another wish: Could you adapt the jump table so that its elements are not Longs, but instead unsigned 32-bit integers. 64-bit numbers are a bit tedious to handle in JS and since we won't request 4 GB mesh data in one request, 32-bit should be enough, I assume.

Reply:
@frcroth nevermind! the front-end can compute the jump table on its own, because it already knows the byteSizes of the individual chunks. I'll remove the jump table code from the back-end next week.